paddle/optimizer/optimizer.cc

```diff
@@ -0,0 +1,75 @@
+#include "optimizer.h"
+#include <string>
+
+#include "parameter_optimizer.h"
+
+template<class T>
+struct EnumToType {};
+
+template<class T>
+struct TypeToEnum {};
+
+#define MATCH_ENUM_TYPE(TYPE, ENUM)                 \
+  template<>                                        \
+  struct TypeToEnum<ENUM> {                         \
+    static paddle_element_type v() {return ENUM;};  \
+    static constexpr TYPE value = ENUM;
+  };
+  template<>                                       \
+  struct EnumToType<ENUM> {                        \
+    typedef TYPE Type;                             \
+  }                                                \
+
+MATCH_ENUM_TYPE(int32_t, PADDLE_ELEMENT_TYPE_INT32);
+MATCH_ENUM_TYPE(uint32_t, PADDLE_ELEMENT_TYPE_UINT32);
+MATCH_ENUM_TYPE(int64_t, PADDLE_ELEMENT_TYPE_INT64);
+MATCH_ENUM_TYPE(uint64_t, PADDLE_ELEMENT_TYPE_UINT64);
+MATCH_ENUM_TYPE(float, PADDLE_ELEMENT_TYPE_FLOAT32);
+MATCH_ENUM_TYPE(double, PADDLE_ELEMENT_TYPE_FLOAT64);
+
+struct paddle_optimizer {
+  /*! \brief optmizer in C++ side */
+
+  paddle::optimizer::ParameterOptimzier* impl;
+};
+
+paddle_optimizer* paddle_create_optimizer(const unsigned char* config_proto,
+                                            int config_proto_len) {
+  paddle_optimizer* optimizer;
+  std::string config(config_proto, config_proto + config_proto_len);
+  optimizer->impl->create(config_proto);
+  return optimizer;
+}
+
+int paddle_release_optimizer(paddle_optimizer* o) {
+  if (o != nullptr)
+    delete o->impl;
```

Comment:
fix the style. Thanks a lot for such a careful review! I would check in my code more carefully, and fix the style, such as go code compile problem by double check.  Thanks !